components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/GeckoLoginStorageDelegate.kt

```diff
@@ -74,11 +75,9 @@ class GeckoLoginStorageDelegate(
         }
     }
 
-    override fun onLoginFetch(domain: String): List<Login> {
-        if (!isAutofillEnabled.invoke()) return listOf()
-        return runBlocking {
-            // GV expects a synchronous response. Blocking here hasn't caused problems during
-            // testing, but we should add telemetry to verify. See #5531
+    override fun onLoginFetch(domain: String): Deferred<List<Login>> {
+        if (!isAutofillEnabled.invoke()) return CompletableDeferred(listOf())
```

Comment:
nit: This was already the case in the previous code... but I think `isAutofillEnabled()` will be easier to read instead of using `invoke()` - which we can omit here. :)